src/components/FormStepper.js

```diff
@@ -0,0 +1,151 @@
+        mapFormStateToFields={mapFormStateToFields} mapFieldsToFormState={mapFieldsToFormState}/>
+    </Route>
+  )
+}
+
+const getCurrentStep = (baseUrl, pathname, stepPaths) => {
+  for (const [stepNumber, stepPath] of stepPaths.entries()) {
+    if (pathname.startsWith(`${baseUrl}/${stepPath}`)) {
+      return {currentStepPath: stepPath, currentStepNumber: stepNumber}
+    }
+  }
+  return {}
+}
+
+const getAdjacentStepUrls = (baseUrl, currentStepPath, stepPaths) => {
+  let previousStepPath = undefined
+  for (const [stepNumber, stepPath] of stepPaths.entries()) {
+    let nextStepPath = stepNumber + 1 >= stepPaths.length ? undefined : stepPaths[stepNumber + 1]
+    if (stepPath === currentStepPath) {
+      return {
+        previousStepUrl: previousStepPath ? `${baseUrl}/${previousStepPath}` : undefined,
+        nextStepUrl: nextStepPath ? `${baseUrl}/${nextStepPath}` : undefined
+      }
+    }
+    previousStepPath = stepPath
+  }
+  return {}
+}
+
+const FormStepper = ({ initialFormState, onFormComplete, t, children }) => {
+  const { url, path } = useRouteMatch();
+  const [formState, setFormState] = useState(initialFormState);
+  const history = useHistory();
+  const { pathname } = useLocation();
+  const stepPaths = React.Children.map(children, child => child.props.stepPath)
+  const {currentStepPath, currentStepNumber} = getCurrentStep(url, pathname, stepPaths)
+  const {previousStepUrl, nextStepUrl} = getAdjacentStepUrls(url, currentStepPath, stepPaths)
+  const onComplete = nextStepUrl
+    ? (formStateUpdates) => {
+        setFormState(formState => ({...formState, ...formStateUpdates}))
+        history.push(nextStepUrl)
+      }
+    : (formStateUpdates) => {
+        setFormState(formState => ({...formState, ...formStateUpdates}))
+        const finalFormState = {...formState, ...formStateUpdates}
+        onFormComplete(finalFormState)
+      }
+  const onBack = previousStepUrl ? () => history.push(previousStepUrl) : undefined
+  const stepperBarSteps = React.Children.map(children, child => ({
+    title: child.props.title,
+    onClick: () => history.push(`${url}/${child.props.stepPath}`)
+  }))
+  return (
+    <Box tag="section" basis="auto">
+      <Disclaimer />
+      <Heading tag="h1">{t('powerOfAttorney')}</Heading>
+      <div>
+        <div className="stepper">
+          <Stepper
+            steps={stepperBarSteps}
+            activeColor="#679ba1"
+            completeColor="#679ba1"
+            activeBorderColor="#679ba1"
+            activeStep={currentStepNumber}
+          />
+        </div>
+      </div>
+      <Paragraph className="align-center">
+        <strong>{t('partI')} </strong>
+        {t('thisFormIsToBeFilled')}
+      </Paragraph>
+      <Box direction="row" justify="center">
+        <Box width="large">
+          <Switch>
+            {React.Children.map(children,
+              child => createStepRoute(path, formState, t, onComplete, onBack, child))}
```

Comment:
I like this. What happens if someone navigates to the middle of the form without filling out prior steps?

Reply:
From the conversation in Slack:

>For blocking navigation to a step past the last completed one there's a few ways we could do it - a strategy that I think would work pretty well is to track the completed steps in FormStepper by adding another line to the onComplete/onBack it passes to each step that would set/unset a completed flag in a bit of state FormStepper tracks.

We decided that we'll mark this as a follow-on improvement for after we get the steps working. I'm opening an issue to track it.